sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningDeserializer.java

```diff
@@ -179,56 +182,66 @@ private void handleFlatteningForField(AnnotatedField annotatedField, JsonNode js
             }
 
             if ((classHasJsonFlatten || annotatedField.hasAnnotation(JsonFlatten.class))
-                && containsFlatteningDots(jsonPropValue)) {
+                && IS_FLATTENED_PATTERN.matcher(jsonPropValue).matches()) {
                 // The jsonProperty value contains flattening dots, uplift the nested
                 // json node that this value resolving to the current level.
-                JsonNode childJsonNode = findNestedNode(jsonNode, jsonPropValue);
+                String[] jsonNodeKeys = Arrays.stream(SPLIT_KEY_PATTERN.split(jsonPropValue))
+                    .map(FlatteningDeserializer::unescapeEscapedDots)
+                    .toArray(String[]::new);
 
-                // This should replace the JsonNode it is compacting.
-                ((ObjectNode) jsonNode).set(jsonPropValue, childJsonNode);
-            }
-        }
-    }
+                // Keep track of the JsonNodes which lead to the flattened property being deserialized.
+                //
+                // This will be used later to clean up the parent JsonNode so that there aren't additional dangling
+                // JSON once the flattened value is uplifted. If this isn't done it can result in models with
+                // @JsonAnySetter containing errant values.
```

Comment:
I'll add a more detailed comment. The following is a class and JSON string I'll use as an example:

```java
public class AClass {
  @JsonFlatten
  @JsonProperty("flattened.string")
  private String flattenedString;

  @JsonIgnore
  private Map<String, Object> additionalProperties;

  @JsonAnySetter
  void addAdditionalProperty(String key, Object value) {
    if (additionalProperties == null) {
      additionalProperties = new HashMap<>();
    }
    additionalProperties.put(key, value);
  }
}
```

```json
{
  "flattened": {
    "string": "string"
  },
  "additionalProperty1": "value1",
  "additionalProperty2": "value2"
}
```

Before this change, after applying the un-flattening of `flattened.string` the JSON tree structure would have been equivalent to the following JSON:

```json
{
  "flattened": {
    "string": "string"
  },
  "additionalProperty1": "value1",
  "additionalProperty2": "value2",
  "flattened.string": "string"
}
```

With the `JsonAnySetter` existing this would have then resulted in an errant entry existing in `additionalProperties` of:

```
additionalProperties.get("flattened") => map of "string":"string"
```

After this change, after applying the un-flattening of `flattened.string` the JSON tree structure leading to the flattened value will be cleaned up for every sub-tree which is now empty, resulting the following JSON:

```json
{
  "additionalProperty1": "value1",
  "additionalProperty2": "value2",
  "flattened.string": "string"
}
```

So, there will no longer be any errant properties existing in `additionalProperties`.